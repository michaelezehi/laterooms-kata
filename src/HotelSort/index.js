import React, { Component } from 'react';
import { func } from 'prop-types';

import { FormGroup, ControlLabel, FormControl } from 'react-bootstrap';

class HotelSort extends Component {

  static propTypes = {
    sortResults: func.isRequired,
  }

  handleSort(e) {
    this.props.sortResults(e.target.value);
  }

  render() {
    return (
      <FormGroup controlId="hotelSort">
        <ControlLabel>Sort:</ControlLabel>
        <FormControl
          componentClass="select"
          onChange={this.handleSort.bind(this)}
        >
          <option value="StarRating|desc">Star Rating Desc</option>
          <option value="StarRating|asc">Star Rating Asc</option>
        </FormControl>
      </FormGroup>
    );
  }
}

export default HotelSort;
