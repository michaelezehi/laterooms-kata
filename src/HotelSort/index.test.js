import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import HotelSort from './index';

import { HOTELS } from '../hotelData.js';

// TODO - move into mocks
const sortResults = (sortQuery = 'StarRating|desc') => {
  const sortedResults = HOTELS;
  const sortVals = sortQuery.split('|');

  sortedResults.sort((a, b) => {
    if (sortVals[1] === 'asc') {
      return a[sortVals[0]] - b[sortVals[0]];
    }

    if (sortVals[1] === 'desc') {
      return b[sortVals[0]] - a[sortVals[0]];
    }
  });

  return sortedResults;
};

it('renders without crashing', () => {
  const component = renderer.create(
    <HotelSort sortResults={sortResults} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('sorts results', () => {
  const component = shallow(
    <HotelSort sortResults={sortResults} />,
  );

  expect(HOTELS[0].Name).toBe('hotelone');
  component.instance().handleSort({ target: 'StarRating|asc' });
  expect(HOTELS[1].Name).toBe('hoteltwo');

});
