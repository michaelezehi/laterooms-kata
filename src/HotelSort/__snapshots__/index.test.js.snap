// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label"
    htmlFor="hotelSort"
  >
    Sort:
  </label>
  <select
    className="form-control"
    id="hotelSort"
    onChange={[Function]}
    type={undefined}
  >
    <option
      value="StarRating|desc"
    >
      Star Rating Desc
    </option>
    <option
      value="StarRating|asc"
    >
      Star Rating Asc
    </option>
  </select>
</div>
`;
